glean_parser/templates/kotlin.geckoview.jinja2

```diff
@@ -0,0 +1,34 @@
+// -*- mode: kotlin -*-
+
+/*
+ * AUTOGENERATED BY glean_parser.  DO NOT EDIT.
+ */
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("PackageNaming")
+package {{ namespace }}
+
+import {{ glean_namespace }}.private.HistogramBase // ktlint-disable import-ordering no-unused-imports
+
+/*
+ * This class performs the mapping between Gecko Histograms and Glean SDK
+ * metric types.
+ */
+internal object GleanGeckoHistogramMapping {
+    // Support exfiltration of Gecko metrics from products using both the
+    // Glean SDK and GeckoView. See bug 1566356 for more context.
+    operator fun get(geckoMetricName: String): HistogramBase? {
+        return when (geckoMetricName) {
+        {% for category in gecko_metrics.keys()|sort %}
+            // From {{ category|Camelize }}.kt
+            {% for metric in gecko_metrics[category] %}
+            "{{ metric.gecko_datapoint }}" -> {{ category|Camelize }}.{{ metric.name|camelize }}
```

Comment:
I'm not sure it matters, but I think this means all of the (Gecko) histograms will be instantiated when you use the first one.

Reply:
Mh, I tried this (check [here](https://pl.kotl.in/cnrMCzHIV) for a PoC) and doesn't look like it's triggering the load of other histograms. Is that what you meant?

Reply:
Yep -- that's what I meant.  Thanks for confirming